pkg/config/config.go

```diff
@@ -0,0 +1,96 @@
+package config
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/kelseyhightower/envconfig"
+	"gopkg.in/yaml.v2"
+)
+
+// OIDCConfig ...
+type OIDCConfig struct {
+	Address string `yaml:"address"`
+
+	ClusterName            string   `yaml:"clusterName" envconfig:"cluster_name"`
+	IssuerURL              string   `yaml:"issuerURL" envconfig:"issuer_url"`
+	ClientID               string   `yaml:"clientID" envconfig:"client_id"`
+	ClientSecret           string   `yaml:"clientSecret" envconfig:"client_secret"`
+	AllowEmptyClientSecret bool     `yaml:"allowEmptyClientSecret" envconfig:"allow_empty_client_secret"`
+	RedirectURL            string   `yaml:"redirectURL" envconfig:"redirect_url"`   // http://gangway.auth.app.local:9080
+	RedirectPath           string   `yaml:"redirectPath" envconfig:"redirect_path"` // /callback
+	Scopes                 []string `yaml:"scopes" envconfig:"scopes"`
+	UsernameClaim          string   `yaml:"usernameClaim" envconfig:"username_claim"`
+	EmailClaim             string   `yaml:"emailClaim" envconfig:"email_claim"`
+	ServeTLS               bool     `yaml:"serveTLS" envconfig:"serve_tls"`
+	Audience               string   `yaml:"audience" envconfig:"audience"`
+
+	// TODO :: decide wether should this be use
+	SessionSecurityKey string `yaml:"sessionSecurityKey" envconfig:"SESSION_SECURITY_KEY"`
+}
+
+// NewConfig returns a Config struct from serialized config file
+func NewConfig(configFile string) (*OIDCConfig, error) {
+
+	cfg := &OIDCConfig{
+		Address:                ":9001",
+		IssuerURL:              "http://dex.auth.app.local:9080",
+		RedirectURL:            "http://192.168.1.134:9500",
+		RedirectPath:           "/callback",
+		AllowEmptyClientSecret: false,
+		Scopes:                 []string{"openid", "profile", "email", "offline_access"},
+		UsernameClaim:          "nickname",
+		EmailClaim:             "",
+		ServeTLS:               false,
+		ClientID:               "bender",
+		ClientSecret:           "googledocs",
+	}
+
+	if configFile != "" {
```

Comment:
The defaults won't work in practice, so we should require a file path here.

Reply:
make it mandatory makes more sense ...